deploy/run-shadow-rehearsal.py

```
@@ -0,0 +1,697 @@
+        if not group_absent or not detached_absent:
+            _fail("shadow callback descendants could not be terminated")
+        if failure:
+            _fail(failure)
+        if group_remained or detached:
+            _fail("shadow callback left descendant processes")
+
+
+def _marked_callback_process_ids(callback_run_id: str) -> set[int]:
+    marker = f"{CALLBACK_RUN_ENV}={callback_run_id}"
+    result = subprocess.run(
+        ["ps", "eww", "-axo", "pid=,command="],
```

Ruff (0.16.3)

[error] Starting a process with a partial executable path

(S607)